File: 792-binary-search/binary-search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function(nums, target) {
    return binarySearch(nums, 0, nums.length-1, target);
};

var binarySearch = (arr, start, end, target) => {
    if(start > end) return -1;
    let mid = Math.floor((start + end)/2);
    if(arr[mid] === target) {
        return mid;
    } else if (arr[mid] > target) {
        return binarySearch(arr, start, mid - 1, target);
    } else {
        return binarySearch(arr, mid + 1, end, target);
    }
}

File: 792-binary-search/binary-search.py
import math 

class Solution:
    def binarySearch(self, nums, left, right, target):
        if left > right:
            return -1
        mid = math.floor(left + ((right - left)/2))

        if nums[mid] == target:
            return mid
        if nums[mid] > target:
            return self.binarySearch(nums, left, mid-1, target)
        else:
            return self.binarySearch(nums, mid+1, right, target)

    def search(self, nums: List[int], target: int) -> int:
        return self.binarySearch(nums, 0, len(nums)-1, target)
    